### PR TITLE
#50 Added ability to choose ALB destination port

### PR DESCRIPTION
**Pull request by TysonBeck**
Added variable and logic for using either HTTP or HTTPS for ALB target group

### CODE REVIEW

File: aws/cerberus/cerberus.tf [hunk 1]
```diff
@@ -99,6 +99,11 @@ variable "sftp_health_check_threshold" {
   default     = "2"
 }
 
+variable "alb_target_group_port" {
+  description = "EC2 target group port from ALB. Allowed Values:80, 443"
```
Comment (jeremyodle) [inline aws/cerberus/cerberus.tf]: ~~Allowed Values:80, 443~~

File: aws/cerberus/cerberus.tf [hunk 2]
```diff
@@ -108,6 +113,7 @@ locals {
 
   remote_state_acct_key = "${coalesce(var.terraform_remote_state_acct_key, "master/${var.stage}/acct")}"
   remote_state_vpc_key  = "${coalesce(var.terraform_remote_state_vpc_key, "master/${var.stage}/shared-vpc")}"
+  alb_protocol          = "${var.alb_target_group_port == 80 ? "HTTP" : "HTTPS" }"
```
Comment (jeremyodle) [inline aws/cerberus/cerberus.tf]: The protocol should be mapped to a variable, but default to `HTTP` (e.g. `alb_target_group_protocol`).

File: aws/cerberus/cerberus.tf [hunk 3]
```diff
@@ -99,6 +99,11 @@ variable "sftp_health_check_threshold" {
   default     = "2"
 }
 
+variable "alb_target_group_port" {
+  description = "EC2 target group port from ALB. Allowed Values:80, 443"
+  default     = "443"
```
Comment (jeremyodle) [inline aws/cerberus/cerberus.tf]: This should default to `80`.

File: aws/cerberus/cerberus.tf [hunk 4]
```diff
@@ -166,6 +172,14 @@ resource "aws_security_group" "cerberus_ec2" {
     description     = "${var.desc_prefix} HTTPS from ALB"
   }
 
+  ingress {
```
Comment (jeremyodle) [inline aws/cerberus/cerberus.tf]: Should these ingress ports/protocol be driven from variables?

File: aws/cerberus/cerberus.tf [hunk 5]
```diff
@@ -358,16 +372,16 @@ resource "aws_lb" "cerberus_alb" {
 resource "aws_lb_target_group" "cerberus_alb_target_group" {
   count = "${var.create && var.enable_https ? 1 : 0 }"
 
-  name                 = "${local.module_prefix}-https"
-  port                 = "443"
-  protocol             = "HTTPS"
+  name                 = "${local.module_prefix}-${lower(local.alb_protocol)}"
+  port                 = "${var.alb_target_group_port}"
+  protocol             = "${local.alb_protocol}"
```
Comment (jeremyodle) [inline aws/cerberus/cerberus.tf]: Given the ALB protocol and the target group protocol could differ (HTTPS vs HTTP), you may want to use the variable suggested above `alb_target_group_protocol`.

File: aws/cerberus/cerberus.tf [hunk 6]
```diff
@@ -358,16 +372,16 @@ resource "aws_lb" "cerberus_alb" {
...
   vpc_id               = "${data.terraform_remote_state.vpc.vpc_id}"
   target_type          = "instance"
   deregistration_delay = "300"
 
   health_check {
     path                = "/login"
-    protocol            = "HTTPS"
+    protocol            = "${local.alb_protocol}"
```
Comment (jeremyodle) [inline aws/cerberus/cerberus.tf]: Given the ALB protocol and the target group protocol could differ (HTTPS vs HTTP), you may want to use the variable suggested above `alb_target_group_protocol`.

**Review by jeremyodle — CHANGES_REQUESTED** [review]

File: aws/cerberus/cerberus.tf (hunk 4 above)
Comment (TysonBeck) [reply aws/cerberus/cerberus.tf]: I was first thinking that changing the target group port would need to add another ingress. Relooking at it, it just needs to match the target group port. So yes I believe that can be driven from a variable.

**Review by jeremyodle — APPROVED** [review]

**Pull request merged**